### PR TITLE
Add benchmark suite for canny

### PR DESCRIPTION
**Pull request by abhishekarya286**
## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Gallery example in `./doc/examples` (new features only)
- [x] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by abhishekarya286** [comment]
@stefanv Please review.

**Comment by abhishekarya286** [comment]
Yes, it was @JDWarner. Thanks for reviewing.

File: asv.conf.json [hunk 1]
```diff
@@ -79,10 +80,10 @@
     //     Platform, as in sys.platform. Possible values for the common
     //     cases: 'linux2', 'win32', 'cygwin', 'darwin'.
     //
-    // "exclude": [
+    "exclude": [
     //     {"python": "3.2", "sys_platform": "win32"}, // skip py3.2 on windows
-    //     {"environment_type": "conda", "six": null}, // don't run without six on conda
-    // ],
+         {"environment_type": "conda", "six": null}, // don't run without six on conda
```
Comment (soupault) [inline asv.conf.json]: Could you elaborate on the story behind this?

Comment (jni) [reply asv.conf.json]: @soupault Ah yes, I was there and quite puzzled by this! Basically, we don't need six anymore... But airspeed velocity does!!! 🤦‍♂️

**Comment by jni** [comment]
Thanks @abhishekarya286!

**Pull request merged**